common/locale.go

```diff
@@ -0,0 +1,38 @@
+package common
+
+import (
+	"context"
+	"net/http"
+	"strings"
+)
+
+// LocaleCode the locale code to use to determine the language to use
+const LocaleCode = ContextKey("LocaleCode")
+
+// SetLocaleCode sets the Locale code used to set the language
+func SetLocaleCode(req *http.Request) context.Context {
+	lang := ExtractLangFromSubDomain(req)
+
+	if c, err := req.Cookie("lang"); err == nil && len(c.Value) > 0 {
```

Comment:
I would add a log line here to say that `lang` is coming from the cookie - useful when debugging to know exactly what is being set and from where.

Reply:
Added